Bottom: 7143593195ffe7e2f53621bfff39bc2b40b3376f
Top:    00b49578e6c89b36d490a5c4cae7a71b9e84d88f
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-08 09:16:27 +0800

Refresh of autofs-5.0.7-fix-potential-null-dereference-in-lookup_mount.patch

---

diff --git a/modules/lookup_file.c b/modules/lookup_file.c
index 58663c6..48f5789 100644
--- a/modules/lookup_file.c
+++ b/modules/lookup_file.c
@@ -1131,7 +1131,7 @@ do_cache_lookup:
 				       mapent, ctxt->parse->context);
 	if (ret) {
 		time_t now = time(NULL);
-		int rv = CHE_FAIL;
+		int rv = CHE_OK;
 
 		cache_writelock(mc);
 		me = cache_lookup_distinct(mc, key);
@@ -1139,7 +1139,8 @@ do_cache_lookup:
 			rv = cache_update(mc, source, key, NULL, now);
 		if (rv != CHE_FAIL) {
 			me = cache_lookup_distinct(mc, key);
-			me->status = now + ap->negative_timeout;
+			if (me)
+				me->status = now + ap->negative_timeout;
 		}
 		cache_unlock(mc);
 		return NSS_STATUS_TRYAGAIN;
